pkg/webhook/webhook.go

```diff
@@ -168,7 +169,14 @@ func (wh *WebHook) serve(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func (wh *WebHook) selfRegistration(webhookConfigName string) error {
+func (wh *WebHook) selfRegistration(webhookConfigName string, webhookNamespaceSelectorLabel string) error {
+	namespaceSelector := metav1.LabelSelector{}
```

Comment:
How about using the value of flag `-namespace` instead if it's set?

Reply:
I assume you meant the Spark job namespace that the operator is monitoring. The label needs to be added on that namespace for the webhook to work. Basically there are two scenarios:

1. The `-namespace` flag sets the Spark job namespace to `myNs`. Then the user needs to manually add some label (e.g. `testLabel=stuff`) to the namespace `myNs`. After that the user can install the operator Helm chart by passing this label (I'll create a PR to the chart once this one is merged). Note that if the user forgot to do that during `helm install`, he can also later manually edit the operator-generated MutatingWebhookConfiguration object.

2. The `-namespace` flag is set to empty string, meaning the operator monitors all namespaces. In this case, nothing else needs to be done. Everything remains the same as before because the entire cluster is going to have just one operator deployment. 

In scenario 1, without the labels, things are still going to be fine if only one operator is deployed in the cluster. But as soon as there are two operators deployed, jobs that use the mutating admission webhook is going to fail because two webhooks are going to modify the same Spark pods.

Maybe I should update the documentation with the above?

Reply:
That's right. But when `--namespace` is set, the webhook still handles Spark pods in all namespaces, unless a label is set.
When `--namespace` is not set, the user needs to make sure that he doesn't set the label either.

Reply:
> That's right. But when --namespace is set, the webhook still handles Spark pods in all namespaces, unless a label is set.

That's why I suggested setting the namespace selector here based on the value of `--namespace` so the webhook handles pods in the right namespace.

Reply:
`--namespace` is the namespace where the operator is installed. The label is supposed to identify the Spark job namespace. The `sparkJobNamespace` setting also only sets the name of the namespace, a separate config option for the label is still needed.

Reply:
If there's a way to add a label to a namespace in a Helm chart, that will do. But the only way I know of is to include a namespace resource template file in the chart. That's not ideal because Helm shouldn't explicitly manage namespace creation/deletion.

Reply:
`--namespace` is the namespace the operator watches for Spark applications, not the namespace where the operator is installed. Instead of using a label selector and requiring users to add the same label(s) to the namespace, how about adding some logic in the webhook to filter pods in namespace specified by `--namespace`  if it's set?

Reply:
I see we are not talking about the same namespace. 
In a command such as `helm install operatorchart --namespace myns`, `--namespace` is where the operator is installed.
The namespace you were referring to was this [guy](https://github.com/helm/charts/blob/master/incubator/sparkoperator/templates/spark-operator-deployment.yaml#L61), which is the `sparkJobNamespace` setting in the chart.

But I get what you are saying now. I'll take a look at that.

Reply:
Done. Please take a look.